src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -852,6 +854,90 @@ JNI_METHOD(void, readPath)
     callback->mReadClient = readClient;
 }
 
+JNI_METHOD(void, subscribeToEventPath)
+(JNIEnv * env, jobject self, jlong handle, jlong callbackHandle, jlong devicePtr, jobject eventPathList, jint minInterval,
+ jint maxInterval)
+{
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    DeviceProxy * device = reinterpret_cast<DeviceProxy *>(devicePtr);
+    if (device == nullptr)
+    {
+        ChipLogError(Controller, "No device found");
+        JniReferences::GetInstance().ThrowError(env, sChipDeviceControllerExceptionCls, CHIP_ERROR_INCORRECT_STATE);
+    }
+
+    std::vector<app::EventPathParams> eventPathParamsList;
+    err = ParseEventPathList(eventPathList, eventPathParamsList);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Error parsing Java event paths: %s", ErrorStr(err)));
+
+    app::ReadPrepareParams params(device->GetSecureSession().Value());
+    params.mMinIntervalFloorSeconds   = minInterval;
+    params.mMaxIntervalCeilingSeconds = maxInterval;
+    params.mpEventPathParamsList      = eventPathParamsList.data();
+    params.mEventPathParamsListSize   = eventPathParamsList.size();
+
+    auto callback = reinterpret_cast<ReportCallback *>(callbackHandle);
+
+    app::ReadClient * readClient =
+        Platform::New<app::ReadClient>(app::InteractionModelEngine::GetInstance(), device->GetExchangeManager(),
+                                       callback->mBufferedReadAdapter, app::ReadClient::InteractionType::Subscribe);
+
+    err = readClient->SendRequest(params);
```

Comment:
for subscription, currently we prefer using readClient->SendAutoResubscribeRequest, similar to what we used in darwin, https://github.com/project-chip/connectedhomeip/blob/master/src/darwin/Framework/CHIP/CHIPDevice.mm#L423.
Also we recommend to use cluster state cache, similar to https://github.com/project-chip/connectedhomeip/blob/master/src/darwin/Framework/CHIP/CHIPDevice.mm#L406, which has fine-grained min-event processing and data version cache.

Reply:
I changed to SendAutoResubscribeRequest.
But, because of code reuse (ReportCallback), it is better not to use cache.

Reply:
Without using cluster state cache, you have to add min event in this ReadEvent/SubscribeEvent API that has last received event number for input parameter.
In AndroidCallback.h, you may update ReportCallback's mBufferedReadAdapter with ClusterStateCache, if you see ClusterStateCache, it has mBufferReadCallback inside